File: Chapter04/ReadWriteParquet.r
#Loading a JSON file as a DataFrame
callDetailsDF <- read.df("/home/spark/sampledata/json/cdrs.json","json")
#Writing the DataFrame out as a Parquet
write.parquet(callDetailsDF,"cdrs.parquet")
#Reading Parquet as a DataFrame
callDetailsParquetDF <- read.parquet("cdrs.parquet")
#Data Manipulation of Parquet Data
createOrReplaceTempView(callDetailsParquetDF,"parquetFile")
topCallLocsDF <- sql("select Origin,Dest, count(*) as cnt from calldetails group by Origin,Dest  order by cnt desc")
head(topCallLocsDF)

File: Chapter04/ReadWriteParquet.py
#Reading a JSON file as a DataFrame
callDetailsDF = spark.read.json("/home/spark/sampledata/json/cdrs.json")
# Write the DataFrame out as a Parquet File
callDetailsDF.write.parquet("cdrs.parquet")
# Loading the Parquet File as a DataFrame
callDetailsParquetDF = spark.read.parquet("cdrs.parquet")
# Standard DataFrame data manipulation
callDetailsParquetDF.createOrReplaceTempView("calldetails")
topCallLocsDF = spark.sql("select Origin,Dest, count(*) as cnt from calldetails group by Origin,Dest  order by cnt desc")
